### PR TITLE
Improvement: Refacto group role elements to access control

### LINKED ISSUE #134: Refacto Group and Role elements
There are a lot of duplication between Group/Role elements. The goal is to create a new component to regroup all shared code, to reduce duplication.

Regroup components about Group and Role for the following elements :
- Table
- Tab
- Filter
- Dialog

### PR DESCRIPTION
**Pull request by justinehell**
Resolves #134

### CODE REVIEW

File: src/components/tab-panel/AccessControlTabPanel.vue [hunk 1]
```diff
@@ -52,30 +53,39 @@
...
 const props = defineProps({
   entity: {
     type: Object,
     required: true,
   },
+  accessControlType: {
+    type: String,
+    required: true,
+  },
   type: {
```
Comment (Zorin95670) [inline src/components/tab-panel/AccessControlTabPanel.vue:89]: rename to `subAccessControlType` or another thing but not `type`

File: src/components/tab-panel/AccessControlTabPanel.vue [hunk 2]
```diff
@@ -205,7 +223,7 @@ function getFilters() {
 }
 
 /**
- * Search and display groups.
+ * Search and display groups or roles.
```
Comment (Zorin95670) [inline src/components/tab-panel/AccessControlTabPanel.vue:226]: groups or roles -> access controls

File: src/pages/RolePage.vue [hunk 3]
```diff
@@ -69,15 +69,17 @@
         type="ROLE"
         :entity="role"
       />
-      <groups-tab-panel
+      <access-control-tab-panel
         name="groups"
+        access-control-type="group"
         type="ROLE"
```
Comment (Zorin95670) [inline src/pages/RolePage.vue]: type="Role"
please rework on it to have only 'role' or only 'ROLE' everywhere

File: src/pages/UserPage.vue [hunk 4]
```diff
@@ -79,13 +79,15 @@
       transition-next="jump-down"
       class="bg-grey-1"
     >
-      <groups-tab-panel
+      <access-control-tab-panel
         name="groups"
+        access-control-type="group"
         type="USER"
```
Comment (Zorin95670) [inline src/pages/UserPage.vue]: same here

File: src/components/tab-panel/AccessControlTabPanel.vue (hunk 2 above)
Comment (justinehell) [reply src/components/tab-panel/AccessControlTabPanel.vue:226]: done

File: src/components/tab-panel/AccessControlTabPanel.vue (hunk 1 above)
Comment (justinehell) [reply src/components/tab-panel/AccessControlTabPanel.vue:89]: subAccessControlType -> type
type -> subType

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**